translators/ChatGPTTranslator.js

```diff
@@ -0,0 +1,78 @@
+class ChatGPTTranslator {
+    constructor(model = 'gpt-4o', openApiKey = "") {
+        this.apiKey = openApiKey;
+        this.apiUrl = 'https://api.openai.com/v1/chat/completions';
+        this.model = model;
+    }
+
+    async translate(text, from, to) {
+        if (!text) return '';
+
+        const prompt = (from === 'auto' || from === '')
+            ? `Translate text to ${to}. Only return the translation without any additional text:\n\n${text}`
+            : `Translate text from ${from} to ${to}. Only return the translation without any additional text:\n\n${text}`;
```

Comment:
Let's generate batch requests to translate array of texts with method `translateBatch` and then call this method with single text in `translate` method.

This approach may improve performance dramatically, since Linguist mostly call `translateBatch` method with as many texts as possible.

If we have 1k texts to translate, in current implementation user would sent 1k requests with prompt overhead, but if we will batch texts, linguist will send 50-200 texts in single request and eventually will sent only 5-20 requests.